python/pyarrow/_flight.pyx

```diff
@@ -376,23 +411,23 @@ cdef class FlightClient:
         """Request information about an available flight."""
         cdef:
             FlightInfo result = FlightInfo.__new__(FlightInfo)
+            FlightDescriptor c_descriptor = \
+                FlightDescriptor.unwrap(descriptor)
 
         with nogil:
             check_status(self.client.get().GetFlightInfo(
-                descriptor.descriptor, &result.info))
+                c_descriptor.descriptor, &result.info))
 
         return result
 
     def do_get(self, ticket: Ticket):
         """Request the data for a flight."""
         cdef:
-            # TODO: introduce unwrap
-            CTicket c_ticket
+            Ticket c_ticket = Ticket.unwrap(ticket)
```

Comment:
You already specified `ticket: Ticket` in the method signature. The only thing bad that can happen here is that `ticket` is None.